src/extensions/menu.extension.ts

```diff
@@ -17,8 +20,32 @@ export class Menu {
     options?: {
       level?: number;
       headline?: string;
+      showHelp?: boolean;
+      helpLabel?: string;
+      backLabel?: string;
+      cancelLabel?: string;
+      byeLabel?: string;
     }
   ) {
+    const cacheIsEmpty = !Object.keys(this.optionsCache).length;
+    if (cacheIsEmpty) {
+      // save everything except level & headline
+      this.optionsCache = options;
```

Comment:
At this point, a copy should be passed instead of the reference of the object: 
```
this.optionsCache = {...options}
```

Otherwise, in the following lines not only the properties of `this.optionsCache` will be deleted, but also those of `options` itself (it is one and the same object). This then results in the properties in line 38 no longer being available.

Reply:
You are right :sweat_smile: , good eye!